PSPDFKitPlugin/PSPDFKitPlugin.m

```diff
@@ -1445,4 +1397,229 @@ - (CGRect)flexibleToolbarContainerContentRect:(PSPDFFlexibleToolbarContainer *)c
+        success = [document addAnnotations:@[annotation] options:nil];
+    }
+
+    if (success) {
+        [self.commandDelegate sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_OK]
+                                    callbackId:command.callbackId];
+    } else {
+        NSLog(@"Failed to add annotation.");
+        [self.commandDelegate sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR]
+                                    callbackId:command.callbackId];
+    }
+}
+
+- (void)removeAnnotation:(CDVInvokedUrlCommand *)command {
+    id jsonAnnotation = [command argumentAtIndex:0];
+    NSString *annotationUUID = jsonAnnotation[@"uuid"];
+    if (annotationUUID.length == 0) {
+        NSLog(@"Invalid annotation UUID.");
+        [self.commandDelegate sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR]
+                                    callbackId:command.callbackId];
+    }
+    
+    PSPDFDocument *document = self.pdfController.document;
+    VALIDATE_DOCUMENT(document)
+    BOOL success = NO;
+
+    NSArray<PSPDFAnnotation *> *allAnnotations = [[document allAnnotationsOfType:PSPDFAnnotationTypeAll].allValues valueForKeyPath:@"@unionOfArrays.self"];
+    for (PSPDFAnnotation *annotation in allAnnotations) {
+        // Remove the annotation if the uuids match.
+        if ([annotation.uuid isEqualToString:annotationUUID]) {
+            success = [document removeAnnotations:@[annotation] options:nil];
+            break;
+        }
+    }
+
+    if (success) {
+        [self.commandDelegate sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_OK]
+                                    callbackId:command.callbackId];
+    } else {
+        NSLog(@"Failed to remove annotation.");
+        [self.commandDelegate sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR]
+                                    callbackId:command.callbackId];
+    }
+}
+
+- (void)getAllUnsavedAnnotations:(CDVInvokedUrlCommand *)command {
+    PSPDFDocument *document = self.pdfController.document;
+    VALIDATE_DOCUMENT(document)
+
+    PSPDFDocumentProvider *documentProvider = document.documentProviders.firstObject;
+    NSData *data = [document generateInstantJSONFromDocumentProvider:documentProvider error:NULL];
+    NSDictionary *annotationsJSON = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:NULL];
+    CDVPluginResult *pluginResult;
+    if (annotationsJSON) {
+        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:annotationsJSON];
+    }  else {
+        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:@{@"localizedDescription": @"Failed to get annotations"}];
+    }
+
+    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+}
+
+- (void)applyInstantJSON:(CDVInvokedUrlCommand *)command {
```

Comment:
@radazzouz I noticed the rename and the proposed naming is to my liking however I would like to be changed to `applyInstantJson` if possible to comply to Camel case naming standard we have on Android. Otherwise it will look a bit out of place in our class naming `ApplyInstanJSONAction`.

Reply:
On iOS, it's okay to have capitalized acronyms in method names. See https://pspdfkit.com/api/ios/Classes/PSPDFDocument.html#/c:objc(cs)PSPDFDocument(im)applyInstantJSONFromDataProvider:toDocumentProvider:lenient:error:

So I'd leave it as-is.